packages/idempotency/src/Exceptions.ts

```diff
@@ -0,0 +1,17 @@
+class IdempotencyItemNotFoundError extends Error {
+
+}
+
+class IdempotencyInvalidStatusError extends Error {
+
+}
+
+class IdempotencyItemAlreadyExistsError extends Error{
```

Comment:
When will this error be used? Seems like if the idempotency item already exists (aka, when the condition expression in the put call fails), then dynamo will throw it's own error?

Reply:
It does seem like its not used at the moment - although, I think it would be best to catch the error from the sdk and then throw this error. This way we maintain all sdk specific errors/operations at this level and simply return the IdempotencyItemAlreadyExistsError. That way regardless of the underlying persistence layer the result will be the same in the event of the item already existing in the store

Reply:
I think that makes sense

Reply:
I missed that condition when writing putRecord, have added that now.